### PR TITLE
Prevent M mode triggers affecting D mode ones

### PR DESCRIPTION
**Pull request by timsifive**
This fixes a corner case when chaining triggers, #275.

### CODE REVIEW

File: xml/hwbp_registers.xml [hunk 1]
```diff
@@ -175,6 +175,17 @@
 
             1: While this trigger does not match, it prevents the trigger with
             the next index from matching.
+
+            Because \Fchain affects the next trigger, hardware must zero it in
+            M mode writes to \Rmcontrol if the next trigger has \Fdmode of 1.
+            In addition hardware should ignore writes to \Rmcontrol with
+            \Fdmode of 1 if the previous trigger has both \Fdmode of 0 and
```
Comment (aswaterman) [inline xml/hwbp_registers.xml]: I would phrase this as "writes to \Rmcontrol that would result in \Fdmode being 1".  As written, it's unclear whether it's talking about the old or new value of \Fdmode.

File: xml/hwbp_registers.xml [hunk 2]
```diff
@@ -175,6 +175,17 @@
 
             1: While this trigger does not match, it prevents the trigger with
             the next index from matching.
+
+            Because \Fchain affects the next trigger, hardware must zero it in
+            M mode writes to \Rmcontrol if the next trigger has \Fdmode of 1.
```
[the remark this answers is not in the document]
Comment (timsifive) [reply xml/hwbp_registers.xml]: Agreed.

File: xml/hwbp_registers.xml (hunk 1 above)
Comment (timsifive) [reply xml/hwbp_registers.xml]: How about "writes to \Rmcontrol that set \Fdmode to 1"?

Comment (aswaterman) [reply xml/hwbp_registers.xml]: yeah

**Pull request merged**